workspace/__lib__/xod-dev/esp8266-mcu/socket/patch.cpp

```diff
@@ -0,0 +1,26 @@
+
+// clang-format off
+{{#global}}
+#include <ESP8266WiFi.h>
+{{/global}}
+// clang-format on
+
+using Type = WiFiClient*;
+
+struct State {
+    Type client;
+};
+
+// clang-format off
+{{ GENERATED_CODE }}
+// clang-format on
+
+void evaluate(Context ctx) {
+    auto state = getState(ctx);
+
+    if (isSettingUp()) {
+        state->client = new WiFiClient();
```

Comment:
Can we use placement new instead? It will make RAM consumption report more accurate. That is, the `State` might only contain `uint8_t mem[sizeof(WiFiClient)]`.

Reply:
The way it is used now, `evaluate` will only be called if someone forgot to link a "proper" `socket`(created with `open-tcp`). So if we use placement new, `mem` will almost always be unused.

Reply:
Ah… that’s a “bad” client, got it. OK.

Reply:
Can you leave a comment in the code explaining that should not normally happen?